.github/workflows/test-and-zip-default-themes.yml

```diff
@@ -49,6 +49,51 @@ concurrency:
 permissions: {}
 
 jobs:
+  # Checks for zero-byte files.
+  #
+  # Occasionally, binary files such as images and fonts are added to themes incorrectly.
+  # This checks that all files contain contents.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Check for zero-byte files.
+  check-for-empty-files:
+    name: Check ${{ matrix.theme }} for empty files
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    timeout-minutes: 10
+    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        theme: [
+          'twentytwentyfour',
+          'twentytwentythree',
+          'twentytwentytwo',
+          'twentytwentyone',
+          'twentytwenty',
+          'twentynineteen',
+          'twentyseventeen',
+          'twentysixteen',
+          'twentyfifteen',
+          'twentyfourteen',
+          'twentythirteen',
+          'twentytwelve',
+          'twentyeleven',
+          'twentyten'
+        ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          ref: ${{ github.event_name == 'workflow_dispatch' && inputs.branch || github.ref }}
+          show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+
+      - name: Find zero-byte files
+        run: find src/wp-content/themes/${{ matrix.theme }} -size 0
```

Comment:
Something like this may work:

```sh
if [[ "" == $(find  src/wp-content/themes/${{ matrix.theme }} -size 0) ]]; then exit 1; fi
```
Not sure if that translates to the workflow properly, but you get the gist :slightly_smiling_face:

Reply:
Ah! I didn't see this before making 7f5fb21dd15583893ccad63a09cae4a1890ffaf2. What do you think about that approach? I think it reads a lot easier. But not opposed to something like above either.

Reply:
The other approach works but I'm not sure that it'll be clear that the diff check is part of, and required for, the empty file checks.

`This also works in bash and may work here:
```yaml
- name: Make sure there are no empty files.
-run: [[ ! $(find . -type f -empty) ]]
```
Do you think that's readable enough or not quite?

Reply:
Hmm, I've added this, but it doesn't seem to be flagging the empty images (I haven't synced my PR branch since that was fixed in `trunk`).

Reply:
Hmm, it should be producing a 1 exit code. Strange. Unless a readable oneliner comes to mind, the diff check makes most sense

Reply:
Strange! I just tried it again and it did work. 🤷‍♂️